<launch>

    <arg name="publish_robot_source_odom_tf" default="true"/>

    <!--TF STATIC-->
    <!-- odom related-->
    <!--move_base-->
    <include file="$(find rubber_navigation)/launch/move_base.launch"/>

    <include file="$(find rubber_navigation)/launch/common.launch">
        <arg name="publish_robot_source_odom_tf" value="$(arg publish_robot_source_odom_tf)"/>
    </include>

    <include file="$(find sick_tim)/launch/sick_tim551_2050001.launch" />

    <include file="$(find rubber_navigation)/launch/amcl.launch"/>

    <node name="goalSaver" pkg="rubber_navigation" type="goalSaver" required="true">
        <param name="map_frame" value="map"/>
        <param name="base_footprint_frame" value="base_link"/>
        <param name="file_name" value="$(find rubber_navigation)/config/treeTarget.txt"/>
    </node>
    <!-- map related-->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rubber_navigation)/config/rviz/amcl.rviz"/>

    <node name="simpleNav" pkg="rubber_navigation" type="simpleNav" output="screen">
        <param name="base_foot_print" value="base_link"/>
        <param name="odom_frame" value="odom"/>
        <param name="publish_tf" value="$(arg publish_robot_source_odom_tf)"/>
        <param name="max_linear_velocity" value="0.8"/>
        <param name="max_angular_velocity" value="0.5"/>
        <param name="serial_addr" value="/dev/ttyUSB0"/>
    </node>
</launch>